tofu/spectro/_rockingcurve.py

```diff
@@ -0,0 +1,345 @@
+    with the perfect and mosaic crystal models, the reflectivity curve with the
+    full dynamical model for parallel and perpendicular photon polarizations.
+
+    The alpha-Quartz, symmetry group D(4,3), is assumed left-handed.
+    It makes a difference for certain planes, for example between
+    (h,k,l)=(2,0,3) and (2,0,-3).
+    The alpha-Quartz structure is hexagonal with 3 molecules of SiO2 in the
+    unit cell.
+
+    All crystal lattice constants and wavelengths are in Angstroms (1e-10 m).
+
+    Parameters:
+    -----------
+    ih, ik, il:    int
+        Miller indices of crystal used
+    lamb:   float
+        Wavelength of interest, in Angstroms (1e-10 m)
+    """
+
+    # Check inputs
+    # ------------
+
+    lc = [ih is not None, ik is not None, il is not None, lamb is not None]
+    if any(lc) and not all(lc):
+        msg = (
+            "Args h, k, l and lamb must be provided together:\n"
+            + "\t - h: first Miller index ({})\n".format(ih)
+            + "\t - k: second Miller index ({})\n".format(ik)
+            + "\t - l: third Miller index ({})\n".format(il)
+            + "\t - lamb: wavelength of interest ({})\n".format(lamb)
+        )
+        raise Exception(msg)
+
+    # Calculations of main crystal parameters
+    # ---------------------------------------
+
+    # Classical electronical radius, in Angstroms
+    re = 2.82084508e-5
+
+    # From Ralph W.G. Wyckoff, "Crystal Structures" (1948)
+    # https://babel.hathitrust.org/cgi/pt?id=mdp.39015081138136&view=1up&seq=259&skin=2021
+    # Inter-atomic distances into hexagonal cell unit (page 239 & 242) and
+    # calculation of the associated volume
+    # TBC with Francesca
+    a0 = 4.9130
+    c0 = 5.4045
+    V = a0**2.*c0*np.sqrt(3.)/2.
+
+    # Atomic number of Si and O atoms
+    Zsi = 14.
+    Zo = 8.
+
+    # Position of the three Si atoms in the unit cell (page 242 Wyckoff)
+    u = 0.4705
+    xsi = np.r_[-u, u, 0.]
+    ysi = np.r_[-u, 0., u]
+    zsi = np.r_[1./3., 0., 2./3.]
+
+    # Position of the six O atoms in the unit cell (page 242 Wyckoff)
+    x = 0.4152
+    y = 0.2678
+    z = 0.1184
+    xo = np.r_[x, y-x, -y, x-y, y, -x]
+    yo = np.r_[y, -x, x-y, -y, x, y-x]
+    zo = np.r_[z, z+1./3., z+2./3., -z, 2./3.-z, 1./3.-z]
+
+    # Bragg angle and atomic plane distance d for a given wavelength lamb and
+    # Miller index (h,k,l)
+    d_num = np.sqrt(3.)*a0*c0
+    d_den = np.sqrt(4.*(ih**2 + ik**2 + ih*ik)*c0**2 + 3.*il**2*a0**2)
+    if d_den == 0.:
+        msg = (
+            "Something went wrong in the calculation of d, equal to 0!\n"
+            "Please verify the values for the following Miller indices:\n"
+            + "\t - h: first Miller index ({})\n".format(ih)
+            + "\t - k: second Miller index ({})\n".format(ik)
+            + "\t - l: third Miller index ({})\n".format(il)
+        )
+        raise Exception(msg)
+    else:
+        d_atom = d_num/d_den
```

Comment:
you don't need `else` here, because if there was an error, the code already stopped